src/se/vidstige/jadb/AdbFilterInputStream.java

```diff
@@ -1,12 +1,21 @@
 package se.vidstige.jadb;
 
+import java.io.BufferedInputStream;
 import java.io.FilterInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 
 public class AdbFilterInputStream extends FilterInputStream {
-    public AdbFilterInputStream(InputStream inputStream) {
-        super(inputStream);
+
+    private Transport mTransport;
```

Comment:
Reason why I added this is because the Transport class opens both an input and output stream, but in here, we just pass the input stream and close that only. I can still remove if I missed the handling the Transport elsewhere.